actors/builtin/miner/miner_actor.go

```diff
@@ -1653,141 +1653,59 @@ func handleProvingDeadline(rt Runtime) {
 
 	hadEarlyTerminations := false
 
-	powerDelta := PowerPair{big.Zero(), big.Zero()}
+	powerDeltaTotal := NewPowerPairZero()
 	penaltyTotal := abi.NewTokenAmount(0)
-	pledgeDelta := abi.NewTokenAmount(0)
+	pledgeDeltaTotal := abi.NewTokenAmount(0)
 
 	var st State
 	rt.State().Transaction(&st, func() {
-		var err error
+		penaltyTarget := abi.NewTokenAmount(0)
+
 		{
 			// Vest locked funds.
 			// This happens first so that any subsequent penalties are taken
 			// from locked vesting funds before funds free this epoch.
 			newlyVested, err := st.UnlockVestedFunds(store, rt.CurrEpoch())
 			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to vest funds")
-			pledgeDelta = big.Add(pledgeDelta, newlyVested.Neg())
+			pledgeDeltaTotal = big.Add(pledgeDeltaTotal, newlyVested.Neg())
 		}
 
 		{
-			// expire pre-committed sectors
-			expiryQ, err := LoadBitfieldQueue(store, st.PreCommittedSectorsExpiry, st.QuantSpecEveryDeadline())
-			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load sector expiry queue")
-
-			bf, modified, err := expiryQ.PopUntil(currEpoch)
-			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to pop expired sectors")
-
-			if modified {
-				st.PreCommittedSectorsExpiry, err = expiryQ.Root()
-				builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to save expiry queue")
-			}
-
-			depositToBurn, err := st.checkPrecommitExpiry(store, bf)
+			depositToBurn, err := st.ExpirePreCommits(store, currEpoch)
```

Comment:
This is a no-op now if there are no precommits (well, if there is no precommit deposit). Can we assume that pre-commits always have non-zero deposits?

Reply:
Hmm, it's not my favourite thing. PCD are zero if the block reward expectation is zero, which will happen one day. I think we should load the queue.